web/frontend/src/components/gamepad.vue

```diff
@@ -37,131 +37,188 @@ const curStates = $ref<Record<string, number>>({
 let handle = -1;
 let prevStates: Record<string, number> = {};
 
+let lastError = Date.now();
 const sendEvent = (newEvent: InputController.Event) => {
-  if (enabled) {
-    const req = new InputController.TriggerEventRequest();
-    req.setController('WebGamepad');
-    req.setEvent(newEvent);
-    window.inputControllerService.triggerEvent(req, new grpc.Metadata(), displayError);
+  if (!enabled) {
+    return;
+  }
+  const req = new InputController.TriggerEventRequest();
+  req.setController('WebGamepad');
+  req.setEvent(newEvent);
+  window.inputControllerService.triggerEvent(req, new grpc.Metadata(), (error: ServiceError | null) => {
+    if (error) {
+      const now = Date.now();
```

Comment:
this gives us a decent enough debounce on spamming errors on disconnect. I want to get reification of disconnection down in addition to an app wide refactoring of knowing when disconnected to disable some actions from happening. this might be best for when the TS SDK is real